### ISSUE TITLE
DNSecure for iPadOS - Server Configuration

### ISSUE THREAD

**Issue by infosecguard** [kkk669/DNSecure#59]
I have discovered a repeatable bug with server settings.  After launching the App, the sidebar list is populated with the full default list of DNS servers as expected - for both DoH and DoT providers.

![IMG_3155](https://user-images.githubusercontent.com/126161458/220894876-79e461f8-7421-45bb-9c2c-5afd2f42bf3c.png)

Whichever service provider’s properties are **first** accessed from the DNSecure UI (e.g., Quad9 - DoH or DoT), all other providers that use the same protocol inherit the **same** server properties.  As such, all DoH providers will have the same settings - as will all DoT providers.

Deleting and re-adding a provider, or creating a new provider doesn’t resolve the problem - and neither does editing.  As such, it isn’t possible to switch between different DNS providers.  Removing and re-adding the App restores the default list - but the problem will recur upon choosing a DNS provider.

The problem is repeatable between iPads of the same model.

DNSecure 1.4.1 
iPadOS 16.x
iPad Pro 12.9” M1

**Comment by kkk669** [comment]
Thank you for reporting the problem.

I'm sorry, but I did not fully understand what you were saying. If possible, could you please describe the steps to reproduce the problem?

**Comment by infosecguard** [comment]
![IMG_3159](https://user-images.githubusercontent.com/126161458/221429290-32dbf5b0-3ede-413a-b884-0926b02a85be.png)
![IMG_3160](https://user-images.githubusercontent.com/126161458/221429293-fb1ecf17-afe5-4e9f-b77e-f0ff1656fb1d.png)
![IMG_3161](https://user-images.githubusercontent.com/126161458/221429295-1de81bff-cd1b-447c-9a95-19e751072186.png)

I hope the three screenshots are self explanatory…

1) In this example, the first DNS Service accessed from the DNSecure GUI is Quad9 (DOH).  Here we see the server details/parameters that are expected.

2) When accessing another DNS Service, in this example 1.1.1.1 (DoH), the expected parameters have been replaced with those of Quad9 (DoH).  Similarly, the Quad9 parameters have replaced all other DoH services.

Attempting to modify/correct the DoH server parameters will simultaneously update ALL services.

This issue is consistent between iPad Pro 12.9” (M1 CPU) devices.  If the DNSecure App is deleted and reinstalled - and a different DNS service, such as Google (DoT) is first accessed, the Google server parameters will be populated for ALL DoT service providers.

This issue doesn’t occur with iPads running iPadOS15.x, or iPhone with iOS16 installed.

To be clear this has been extensively tested with multiple devices:

- I have two different iPad Pro 12.9” M1 devices, signed-in to iCloud with entirely different AppleIDs - and BOTH exhibit the same issue when running iPadOS 16.x.  

- I also have access to two other iPad Pros, running iPadOS15.7.2 - and NEITHER of these exhibit this issue.
  
- Similarly, two different iPhones (iPhone 13 Pro and iPhone 14 Pro) with iOS 16.x installed - and NEITHER iPhone have the problem.

In conclusion, the problem appears to exist only with iPad Pro with M-Series CPU.

**Comment by kkk669** [comment]
Thank you for the detailed information. I can reproduce the issue now. I'm going to look into it.

**Comment by kkk669** [comment]
This issue also occurs on macOS.

**Comment by kkk669** [comment]
Hi @infosecguard, I fixed the issue in the latest TestFlight build (1.4.2 (16)). It will be released soon via TestFlight. If you continue to experience problems, please send me feedback.

I will release it to the App Store as well after I test it well.

**Comment by infosecguard** [comment]
DNSecure 1.4.2 (16) fully resolves the issue with iPadOS16.x - fully restoring expected UI behaviour.

Production release recommended.

**Comment by kkk669** [comment]
Thanks for testing. I just submitted 1.4.2 to Apple's App Review.

**Comment by kkk669** [comment]
I've released DNSecure 1.4.2 to the App Store, so I'm closing this issue.
